PORT4324 - gitlab integration docs (#444)
port-labs/port-docs: docs/build-your-software-catalog/sync-data-to-catalog/git/gitlab/installation.md

# Installation

:::note Prerequisites

> - A gitlab account with admin privileges.;

Reviewer: ```suggestion
- A gitlab account with admin privileges;
```

> - A gitlab group account with api privileges. (Refer to the [permissions](./gitlab.md#permissions) section for more information);

Reviewer: ```suggestion
- A gitlab group account with `api` privileges (Refer to the [permissions](./gitlab.md#permissions) section for more information);
```

- A kubernetes cluster to install the integration on;
- Your Port user role is set to `Admin`.

:::

> 1. Sign in to GitLab and go to your desired group's settings.

Reviewer: Since there is an image after every step here, the sentence should end with a colon (`:`) and not a period

   ![GitLab group settings](../../../../../static/img/integrations/gitlab/GitLabGroupSettings.png)

2. In the "Access Tokens" section, click "Create access token."

   ![GitLab group access tokens](../../../../../static/img/integrations/gitlab/GitLabGroupAccessTokens.png)

> 3. Fill in the token details: name, expiration (optional), and select API scope.

Reviewer: ```suggestion
3. Fill in the token details: name, expiration (optional), and select `api` scope.
```

Reviewer: These steps no longer have images, so they should end with a semicolon

> 4. Click "Create group access token."

Reviewer: ```suggestion
4. Click "Create group access token";
```

> 5. Copy the generated token and use it in the integration installation process.

Reviewer: This sentence makes it seem like there is something else the user needs to do before moving forward. I believe the sentence should be changed to:

Copy the generated token and use it when deploying the integration in the following steps.

Or something similar

> 6. Click the ingest button for the blueprint you want to ingest using GitLab.

Reviewer: Mention that this is in the builder page, some users might not know where to find this menu

   ![DevPortal Builder ingest button](../../../../../static/img/integrations/gitlab/DevPortalBuilderIngestButton.png)

7. Select GitLab under the Git providers category.

   ![DevPortal Builder GitLab option](../../../../../static/img/integrations/gitlab/DevPortalBuilderGitLabOption.png)

8. Copy the helm installation command and set the missing parameters.

>    ### tokenMapping

Reviewer: In general this section is pretty important (both of these parameters are). Maybe it would be better to extract them to a separate section in this page so that it is easy to link to them and also give them some more structure

>    GitLab integration support fetching data related to specific paths in your gitlab groups. In order to do so, you need to map the desired paths to the relevant access tokens.

Reviewer: ```suggestion
   The GitLab integration support fetching data related to specific paths in your GitLab groups. The integration is also able to fetch data from different GitLab parent groups by providing additional group tokens. In order to do so, you need to map the desired paths to the relevant access tokens.
```

>    The `tokenMapping` support specifying the path using [globPatterns](https://www.malikbrowne.com/blog/a-beginners-guide-glob-patterns)[] that the integration will search the files in.

Reviewer: ```suggestion
   The `tokenMapping` parameter supports specifying the paths that the integration will search for files and information in, using [globPatterns](https://www.malikbrowne.com/blog/a-beginners-guide-glob-patterns)[].
```

>    Example: `{"*/MyImportantProjects/**": "myImportantProjectsToken", "*/NotImportatn/*": "OtherToken"}`

Reviewer: ```suggestion
   Example: `{"*/MyFirstGitLabProject/**": "<MY_FIRST_GIRLAB_PROJECT_GROUP_TOKEN>", "*/MySecondGitLabProject/*": "<MY_SECOND_GIRLAB_PROJECT_GROUP_TOKEN>"}`
```

Reviewer: This example is a bit too simple, we need to add a few more examples that are a bit more complex, such as providing an array of patterns for a single project

Reviewer: Maybe just that one additional example will be enough for now

   #### appHost

>    Inorder for the GitLab integration to be able to update the data in port on every change in the gitlab repository, you need to specify the `appHost` parameter.

Reviewer: ```suggestion
   In order for the GitLab integration to update the data in Port on every change in the GitLab repository, you need to specify the `appHost` parameter.
```

>    The `appHost` parameter should be set to the url of your gitlab integration instance.

Reviewer: ```suggestion
   The `appHost` parameter should be set to the `url` of your GitLab integration instance. In addition, your GitLab instance (whether it is GItLab SaaS or a self-hosted version of GitLab) needs to have the option to send webhook requests to the Gitlab integration instance, so please configure your network accordingly.
```

> 9. Run the helm command over your kubernetes cluster

Reviewer: ```suggestion
9. Run the helm command with the updated parameters to install the integration in your Kubernetes cluster.
```

> # Self-hosted GitLab

Reviewer: ```suggestion
## Self-hosted GitLab
```
